Use PyO3 functionality to remove Python files (#302)
NNPDF/pineappl: CHANGELOG.md

- added new method `Grid::delete_orders` and the corresponding switch
  `--delete-orders` in the subcommand `write` of the CLI

### Changed

> - renamed `pineappl.lumi.rs` to `pineappl.channel.rs`

Reviewer: ```suggestion
- dropped top-level Python interface layer
- renamed 'lumi' to 'channel' in PyO3 Python interface. This concerns the 1) the argument names of `convolute_with_one` and similar functions; 2) the module `pineappl.lumi.rs` was moved to `pineappl.channel.rs`; 3) the class `LumiEntry` was renamed to `Channel`
```
a) let's be more explicit and b) we should add the changes from #301 as well

> - renamed `pineappl.lumi.rs` to `pineappl.channel.rs`

Reviewer 2: prefix all of your items with `Python API:` to make clear that this concerns it and not the Rust API.

Reviewer: @cschwan correct me if I'm wrong, but I believe what he meant is
```
- PythonAPI: dropped top-level Python interface layer
- PythonAPI: blub
```

Reviewer 2: ... with a space between 'Python' and 'API' though.

> - renamed `pineappl.lumi.rs` to `pineappl.channel.rs`

Reviewer 2: This should probably talk about renaming the modules. What's unclear is that the first `.` refers to a module and the second one to file suffix.

Reviewer: I adjusted my suggested change to reflect that - do you agree with it?

> - removed `.create()` and `.from_grid()` methods and directly instantiate objecs from the class

Reviewer 2: Removed items should be listed under a `### Removed` heading

Reviewer: ```suggestion
- removed `Grid.create()` and `FkTable.from_grid()` methods; use the constructors of the respective classes instead
```
+ I suggest a slight rewording

- `.into()` needs to be explicitly called on subgrids when using `set_subgrid()`

> - replaced `pineappl.grid.PidBasis` with `pineappl.evolution.PidBasis`
> - replaced `pineappl.grid.OperatorSliceInfo` with `pineappl.evolution.OperatorSliceInfo`

Reviewer: ```suggestion
- replaced `pineappl.grid.PyPidBasis` with `pineappl.evolution.PidBasis`
- replaced `pineappl.grid.PyOperatorSliceInfo` with `pineappl.evolution.OperatorSliceInfo`
```

- added new method `Grid::delete_orders` and the corresponding switch
  `--delete-orders` in the subcommand `write` of the CLI

### Changed

- dropped top-level Python interface layer
- renamed `lumi` to `channel` in PyO3 Python interface. This concerns 1) the argument names of `convolute_with_one` and similar functions; 2) the module `Python API:pineappl.lumi` was moved to `Python API:pineappl.channel`; 3) the class `LumiEntry` was renamed to `Channel`
- `.into()` needs to be explicitly called on subgrids when calling `Python API:pineappl.grid.set_subgrid()`

> - replaced `Python API:pineappl.grid.PyPidBasis` with `Python API:pineappl.evolution.PidBasis`

Reviewer 2: sorry, what I meant was something along the lines of

- Python API: replaced `pineappl.grid.PyPidBasis` with `pineappl.evolution.PidBasis`

Reviewer 2: Also manually break the lines after a maximum of 80 characters.

Author: Also addressed in [230b188](https://github.com/NNPDF/pineappl/pull/302/commits/230b188c48df1d817e7dbf949f74d271356de50f).

- added new method `Grid::delete_orders` and the corresponding switch
  `--delete-orders` in the subcommand `write` of the CLI

### Changed

- Python API: dropped top-level Python interface layer
- Python API: renamed `lumi` to `channel` in PyO3 Python interface. This
  concerns 1) the argument names of `convolute_with_one` and similar functions;

>   2) the module `pineappl.lumi` was moved to `pineappl.channel`; 3) the class

Reviewer: ```suggestion
  2) the module `pineappl.lumi` was moved to `pineappl.boc`; 3) the class
```
should we add more details to the changelog?

Author: Yes, that one should definitely be fixed. Maybe @cschwan wants to add more information there?

Reviewer 2: Good catch!
